bcipy/helpers/triggers.py

```diff
@@ -26,72 +47,74 @@ def reset(self):
+                when the calibration trigger is fired; accepts a single
+                parameter for the timing information.
+    Return:
+        timing(array): timing values for the calibration triggers to be written to trigger file or
+                used to calculate offsets.
     """
     trigger_callback = TriggerCallback()
 
-    # If sound trigger is selected, output calibration tones
-    if trigger_type == SOUND_TYPE:
-        play_sound(
-            sound_file_path='bcipy/static/sounds/1k_800mV_20ms_stereo.wav',
-            dtype='float32',
-            track_timing=True,
-            sound_callback=trigger_callback,
-            sound_load_buffer_time=0.5,
-            experiment_clock=experiment_clock,
-            trigger_name='calibration_trigger')
-
-    elif trigger_type == IMAGE_TYPE:
-        if display:
-            calibration_box = visual.ImageStim(
-                win=display,
-                image='bcipy/static/images/testing_images/white.png',
-                pos=(-.5, -.5),
-                mask=None,
-                ori=0.0)
-            calibration_box.size = resize_image(
-                'bcipy/static/images/testing_images/white.png', display.size,
-                0.75)
-
-            display.callOnFlip(trigger_callback.callback, experiment_clock,
-                               trigger_name)
-            if on_trigger is not None:
-                display.callOnFlip(on_trigger, trigger_name)
-
-            presentation_time = int(1 * display.getActualFrameRate())
-            for _ in range(presentation_time):
-                calibration_box.draw()
-                display.flip()
-
-        else:
-            raise Exception(
-                'Display object required for calibration with images!')
-
-    else:
-        raise Exception('Trigger type not implemented for Calibration yet!')
-
+    # catch invalid trigger types
+    if trigger_type not in CalibrationType.list():
+        msg = f'Trigger type=[{trigger_type}] not implemented'
+        log.exception(msg)
+        raise BciPyCoreException(msg)
+
+    if not display:
+        msg = f'Calibration type=[{trigger_type}] requires a display'
+        log.exception(msg)
+        raise BciPyCoreException(msg)
+
+    if trigger_type == CalibrationType.IMAGE.value:
+        calibration_box = visual.ImageStim(
+            win=display,
+            image=CALIBRATION_IMAGE_PATH,
+            pos=(-.5, -.5),
+            mask=None,
+            ori=0.0)
+        calibration_box.size = resize_image(CALIBRATION_IMAGE_PATH, display.size, 0.75)
+
+        display.callOnFlip(trigger_callback.callback, experiment_clock, trigger_name)
+        if on_trigger is not None:
+            display.callOnFlip(on_trigger, trigger_name)
+
+        calibration_box.draw()
+        display.flip()
+
+    elif trigger_type == CalibrationType.TEXT.value:
+        calibration_text = visual.TextStim(display, text='')
+
```

Comment:
Everything after this line is essentially repeated in both clauses. We might want to factor the creation of the visual stim into another function: `visual_stim(trigger_type, display) -> visual.BaseVisualStim`

or make `CalibrationType` an Abstract class with subclasses (`TextCalibrationType`, `ImageCalibrationType`) that know how to create a visual stimuli from a display.

```  
    calibration_stim = visual_stim( trigger_type, display)
    display.callOnFlip(trigger_callback.callback, experiment_clock, trigger_name)
    if on_trigger is not None:
        display.callOnFlip(on_trigger, trigger_name)
    calibration_box.draw()
     display.flip()
```

Reply:
I do think we'll need to do something like this. We will likely need more offset corrections throughout the sessions and will need to refactor to support that. I'll hold off for those tickets to implement something like this.